edxval/api.py

```diff
@@ -732,6 +732,49 @@ def get_videos_for_course(course_id, sort_field=None, sort_dir=SortDirection.asc
         pagination_conf,
     )
 
+def get_transcript_details_for_course(course_id):
+    """
+    Returns an iterator of video transcript details given a course id.
+    
+    Args:
+        course_id (String)
+
+    Returns:
+        (dict): Returns all the edx_video_id's and related transcript details for a course
+        {
+        'edx_video_id': {
+            'lang_code': {
+                'provider': 'What the provider is',
+                'file_format': 'file format',
+                'url': 'location of the file',
+                'name': 'name of the file',
+                'size': size of the file
+            }
+        }
+    """
+
+    course_transcripts_data = {}
+
+    course_videos = CourseVideo.objects.filter(course_id=course_id)
+    for course_video in course_videos:
+
+        edx_video_id = course_video.video.edx_video_id
+        transcript_data = {}
+
+        video_transcripts = VideoTranscript.objects.filter(video=course_video.video)
```

Comment:
You can use Django's select_related (for 1-1, FK) and prefetch_related (reverse 1-M, M-M) to optimize DB fetch operations.

- https://docs.djangoproject.com/en/4.2/ref/models/querysets/#select-related
- https://docs.djangoproject.com/en/4.2/ref/models/querysets/#prefetch-related